dali/python/nvidia/dali/_conditionals.py

```diff
@@ -281,15 +283,15 @@ def preprocess_input(self, data_node):
         # otherwise, we need to fill in the splits.
         return self._realize_split(data_node, stack_level)
 
-    def register_data_nodes(self, data_node, global_scope=False):
+    def register_data_nodes(self, data_nodes, global_scope=False):
         """Register the data nodes as produced in current scope, otherwise if `global_scope` is True
         put them in the outermost scope.
         """
         if not self._is_registration_allowed:
             return
-        logging.log(8, (f"{self._indent()}[IF/Register] {data_node} at {self.stack_depth() -1}"))
+        logging.log(8, (f"{self._indent()}[IF/Register] {data_nodes} at {self.stack_depth() -1}"))
         scope = self._stack[0] if global_scope else self.top()
-        scope.add_produced(data_node)
+        tree.map_structure(lambda node: scope.add_produced(node), data_nodes)
```

Comment:
Btw. I can see that `add_produced` does on its own some kind of list flattening (up to list of the lists). Is that needed?

Reply:
I can check as this might be used in one another place, but maybe untangling it, or moving this map_structure one level higher would also work.